### PR TITLE
Profile Trigger Simplification

### PR DESCRIPTION
**Pull request by mnbf9rca**
## Summary
- Redesign profile configuration to separate "start by" triggers from "continue until" stop conditions
- Replace legacy `blockingStrategyId` system with new `ProfileStartTriggers` and `ProfileStopConditions` models
- Add UI components: StartTriggerSelector, StopConditionSelector, ScheduleTimePicker
- Add sync conflict detection for schema version mismatches between devices
- Add backwards compatibility via `compatibilityStrategyId` for older app versions
- Migrate existing V1 profiles to V2 schema on app launch

## Test Plan
- [ ] Create new profile with manual start + timer stop
- [ ] Create new profile with NFC start + same NFC stop
- [ ] Create new profile with schedule-based start/stop
- [ ] Edit existing V1 profile and verify migration preserves behavior
- [ ] Verify sync between devices with same schema version
- [ ] Verify sync conflict banner appears when syncing with older device
- [ ] Verify start button respects configured triggers
- [ ] Verify stop validation enforces configured conditions

## Summary by Sourcery

Migrate focus profiles from a legacy blocking-strategy system to a new schema-driven trigger model, updating storage, sync, UI, and validation so that profile start and stop behavior is configured explicitly via start/stop triggers with backward compatibility and sync-conflict handling.

New Features:
- Introduce ProfileStartTriggers and ProfileStopConditions models to configure how profiles start and stop, including NFC, QR, schedule, manual, and deep-link options.
- Add StartTriggerSelector, StopConditionSelector, and ScheduleTimePicker UI components for configuring profile triggers and schedules in profile detail screens.
- Add trigger-based start flows from the home screen, including NFC/QR scanning and a start-method picker when multiple triggers are available.
- Add sync conflict detection and user-facing banner when profiles are edited from devices using an older schema version.

Bug Fixes:
- Harden NFC URL parsing to avoid CoreNFC exceptions by restricting URI decoding to well-known type records.
- Improve user-facing error messages for invalid or missing profile links and generic profile-loading errors.

Enhancements:
- Replace legacy blockingStrategyId-driven behavior with a trigger-based start/stop system, including validation of allowed stop methods per profile.
- Validate manual and deep-link stop actions against configured stop conditions to prevent unconfigured or unsafe stops.
- Standardize NFC/QR session tags with type prefixes and reconcile local session start times with server state on sync.
- Persist trigger configuration on BlockedProfiles with schema versioning and JSON-backed storage, including migration helpers and backward-compatible strategy ID mapping.
- Deprecate legacy strategy and physical-unlock UI components in favor of the new trigger-based configuration model and validation.
- Add NFC usage description to Info.plist and make NFCScannerUtil observable for reuse across views.
- Refine deep-link start behavior to ensure profiles are only started when configured for deep-link triggers and avoid leaving users without an active session on failed switches.

Documentation:
- Document simulator-start performance considerations when running the iOS test suite in AGENTS.md.

Tests:
- Add comprehensive unit tests covering trigger models, migration from legacy strategies, trigger validation, start/stop action determination, sync conflict handling, and profile schema migration behavior.

### CODE REVIEW

**Comment by mnbf9rca** [comment]
@sourcery-ai review

File: Foqos/Views/BlockedProfileView.swift [hunk 1]
```diff
@@ -849,6 +850,12 @@ struct BlockedProfileView: View {
...
                 )
 
+                // Save trigger configuration to profile
+                triggerConfig.saveToProfile(newProfile)
+
+                // Update legacy strategy ID for backwards compatibility with older devices
+                newProfile.updateCompatibilityStrategyId()
+
                 // Schedule restrictions
                 DeviceActivityCenterUtil.scheduleTimerActivity(for: newProfile)
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Views/BlockedProfileView.swift]: Fixed: Added validation check before save with bulleted error formatting.

File: Foqos/Views/HomeView.swift [hunk 2]
```diff
@@ -327,12 +370,61 @@ struct HomeView: View {
...
+    case .startImmediately:
+      strategyManager.toggleBlocking(context: context, activeProfile: profile)
+
+    case .scanNFC:
+      // TODO: Show NFC scanner - for now fall back to toggle
+      strategyManager.toggleBlocking(context: context, activeProfile: profile)
+
+    case .scanQR:
+      // TODO: Show QR scanner - for now fall back to toggle
+      strategyManager.toggleBlocking(context: context, activeProfile: profile)
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Views/HomeView.swift]: Verified: No TODO comments exist; scanner integration is complete with StartNFCScannerSheet and StartQRScannerSheet.

File: Foqos/Views/HomeView.swift [hunk 3]
```diff
@@ -319,6 +335,33 @@ struct HomeView: View {
...
+    case .startImmediately:
+      return "Start Now"
+    case .scanNFC:
+      return "Scan NFC Tag"
+    case .scanQR:
+      return "Scan QR Code"
+    case .waitForSchedule:
+      return "Wait for Schedule"
+    case .showPicker:
+      return "Choose Method"
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Views/HomeView.swift]: By design: 'Choose Method' is the intended dialog label when picker is shown.

File: Foqos/CloudKit/SyncCoordinator.swift [hunk 4]
```diff
@@ -150,8 +150,15 @@ class SyncCoordinator: ObservableObject {
...
+          // Check schema version - ignore changes from older schema versions
+          if syncedProfile.profileSchemaVersion < existingProfile.profileSchemaVersion {
+            Log.warning(
+              "Ignoring sync from older schema version for profile: \(existingProfile.name)",
+              category: .sync
+            )
+            SyncConflictManager.shared.addConflict(profileId: existingProfile.id)
+          } else if syncedProfile.version > existingProfile.syncVersion {
+            // Update existing profile if remote version is newer and schema is same or newer
             updateLocalProfile(existingProfile, from: syncedProfile, in: context)
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/CloudKit/SyncCoordinator.swift]: Fixed: Added clearConflict() call after successful profile update.

File: Foqos/Models/ProfileStartTriggers.swift [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+/// Defines which triggers can start a blocking session for a profile.
+/// Multiple triggers can be enabled simultaneously.
+struct ProfileStartTriggers: Codable, Equatable {
+  var manual: Bool = false
+  var anyNFC: Bool = false
+  var specificNFC: Bool = false
+  var anyQR: Bool = false
+  var specificQR: Bool = false
+  var schedule: Bool = false
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/ProfileStartTriggers.swift]: Verified: deepLink is fully implemented with UI toggle and StrategyManager logic.

**Comment by mnbf9rca** [comment]
@sourcery-ai review

File: Foqos/Utils/StrategyManager.swift [hunk 6]
```diff
@@ -1286,3 +1402,172 @@ extension Notification.Name {
...
+  static func determineStartAction(for triggers: ProfileStartTriggers) -> StartAction {
+    var manualOptions: [StartAction] = []
+
+    if triggers.manual {
+      manualOptions.append(.startImmediately)
+    }
+    if triggers.hasNFC {
+      manualOptions.append(.scanNFC)
+    }
+    if triggers.hasQR {
```
Comment (SourceryAI) [inline Foqos/Utils/StrategyManager.swift]: **suggestion (bug_risk):** Deep-link-only profiles are surfaced as "schedule" in the home error messaging, which may confuse users.

In `determineStartAction`, when there are no manual options and `triggers.deepLink` is true, the function returns `.waitForSchedule`. In `HomeView.handleStartTap`, `.waitForSchedule` always shows `"This profile starts on schedule"`, which is inaccurate for deep-link–only profiles.

Consider either:
- Adding a distinct `StartAction` (e.g. `.deepLinkOnly`) and handling it with a dedicated message, or
- Branching in `HomeView` on `profile.startTriggers` when handling `.waitForSchedule` so that deep-link–only profiles show a more accurate error message.

Suggested implementation:

```
enum StartAction: Equatable, Hashable {
  case startImmediately
  case scanNFC
  case scanQR
  case waitForSchedule
  case deepLinkOnly
  indirect case showPicker(options: [StartAction])
}

```

```
extension StrategyManager {
  /// Determines what action to take based on enabled start triggers
  static func determineStartAction(for triggers: ProfileStartTriggers) -> StartAction {
    var manualOptions: [StartAction] = []

    if triggers.manual {
      manualOptions.append(.startImmediately)
    }
    if triggers.hasNFC {
      manualOptions.append(.scanNFC)
    }
    if triggers.hasQR {
      manualOptions.append(.scanQR)
    }

    // If no manual options, fall back to non-manual triggers
    if manualOptions.isEmpty {
      if triggers.schedule {
        return .waitForSchedule
      }
      if triggers.deepLink {
        // Deep-link–only profiles should surface a distinct action
        return .deepLinkOnly
      }

      // Fallback for profiles with no recognizable triggers
      return .waitForSchedule
    }

    // Single manual option → use it directly
    if manualOptions.count == 1, let option = manualOptions.first {
      return option
    }

    // Multiple manual options → show picker
    return .showPicker(options: manualOptions)
  }

      // Validate manual stop for V2 profiles

```

1. Update all `switch` statements on `StartAction` (for example in `HomeView.handleStartTap`) to handle the new `.deepLinkOnly` case. In `HomeView.handleStartTap`, handle `.deepLinkOnly` with a message specific to deep-link–only profiles instead of the schedule message (e.g. `"This profile can only be started from a deep link"` or whatever matches your UX copy).
2. If you have any `default:` cases in switches over `StartAction`, consider replacing them with explicit handling of `.deepLinkOnly` to keep exhaustiveness and avoid masking future additions.
3. If `StartAction` is persisted (e.g., encoded/decoded), ensure any `Codable`/persistence logic is updated to support the new case.

File: Foqos/CloudKit/SyncCoordinator.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/CloudKit/SyncCoordinator.swift]: Already addressed — `clearConflict()` is called after profile updates.

File: Foqos/Utils/TriggerMigration.swift [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+  static func migratePhysicalUnlock(
+    stopConditions: ProfileStopConditions,
+    physicalUnblockNFCTagId: String?,
+    physicalUnblockQRCodeId: String?
+  ) -> (ProfileStopConditions, String?) {
+    var stop = stopConditions
+    var stopTagId: String?
+
+    if let nfcTagId = physicalUnblockNFCTagId {
+      // Replace anyNFC with specificNFC
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Utils/TriggerMigration.swift]: Won't fix — the V1 schema only allows one physical unlock method per profile. Having both NFC and QR IDs set simultaneously is not a valid V1 state. The NFC-first preference is a reasonable defensive default.

File: Foqos/Models/BlockedProfiles.swift [hunk 8]
```diff
@@ -516,3 +635,99 @@ class BlockedProfiles {
...
+        if start.manual && stop.anyQR {
+            return "QRManualBlockingStrategy"
+        }
+        if start.manual && stop.timer && !stop.anyNFC && !stop.anyQR {
+            return "ShortcutTimerBlockingStrategy"
+        }
+
+        // Default to manual
+        return "ManualBlockingStrategy"
+    }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/BlockedProfiles.swift]: Fixed in 0fc1344 — `compatibilityStrategyId` now uses `start.hasNFC`/`start.hasQR` and `stop.hasNFC`/`stop.hasQR` which include `specificNFC`/`specificQR`/`sameNFC`/`sameQR` variants.

File: Foqos/Utils/StrategyManager.swift [hunk 9]
```diff
@@ -1093,6 +1155,78 @@ class StrategyManager: ObservableObject {
...
+  func startWithNFCTag(context: ModelContext, profile: BlockedProfiles, tagId: String) {
+    let prefixedTag = "nfc:\(tagId)"
+    startWithTag(context: context, profile: profile, tag: prefixedTag)
+  }
+
+  /// Start blocking with a pre-scanned QR code (for trigger-based start)
+  func startWithQRCode(context: ModelContext, profile: BlockedProfiles, codeValue: String) {
+    let prefixedTag = "qr:\(codeValue)"
+    startWithTag(context: context, profile: profile, tag: prefixedTag)
+  }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Utils/StrategyManager.swift]: Fixed in 0fc1344 — `startWithNFCTag` and `startWithQRCode` now validate the scanned tag/code against `profile.startNFCTagId`/`startQRCodeId` when `specificNFC`/`specificQR` is enabled. Returns user-facing error on mismatch.

File: Foqos/Models/BlockedProfiles.swift [hunk 10]
```diff
@@ -516,3 +635,99 @@ class BlockedProfiles {
...
+            return "QRTimerBlockingStrategy"
+        }
+        if start.manual && stop.anyQR {
+            return "QRManualBlockingStrategy"
+        }
+        if start.manual && stop.timer && !stop.anyNFC && !stop.anyQR {
+            return "ShortcutTimerBlockingStrategy"
+        }
+
+        // Default to manual
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/BlockedProfiles.swift]: Won't fix — the current ordering is intentional. NFC/QR strategies are more restrictive and take precedence. V1-migrated profiles have exact combinations that map back correctly, and V2 profiles with novel combinations get the closest match for backward compat.

File: Foqos/Models/TriggerConfigurationModel.swift [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+  /// Call when stop conditions change to re-run validation
+  func stopConditionsDidChange() {
+    validate()
+  }
+
+  /// Run validation and update error list
+  func validate() {
+    validationErrors = validator.validate(start: startTriggers, stop: stopConditions)
+    if !validationErrors.isEmpty {
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/TriggerConfigurationModel.swift]: Fixed in 0fc1344 — `validate()` now checks for missing tag IDs when `specificNFC`/`specificQR` are enabled, and missing schedules when schedule toggles are on.

File: Foqos/Components/SyncConflictBanner.swift [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+      }
+
+      Spacer()
+
+      Button {
+        onDismiss()
+      } label: {
+        Image(systemName: "xmark")
+          .foregroundStyle(.secondary)
+      }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Components/SyncConflictBanner.swift]: Fixed in 0fc1344 — added `.accessibilityLabel("Dismiss sync conflict")` to the xmark button.

File: Foqos/Utils/DeviceActivityCenterUtil.swift [hunk 13]
```diff
@@ -11,22 +11,52 @@ class DeviceActivityCenterUtil {
...
+            // For interval end: use V2 stop schedule if available, otherwise use start time + 23:59
+            if profile.stopConditions.schedule,
+               let stopSched = profile.stopSchedule, stopSched.isActive {
+                intervalEnd = DateComponents(hour: stopSched.hour, minute: stopSched.minute)
+            } else {
+                // No scheduled stop — set end to just before start (full day window)
+                let endHour = (startSched.hour + 23) % 24
+                let endMinute = startSched.minute > 0 ? startSched.minute - 1 : 59
+                intervalEnd = DateComponents(hour: endHour, minute: endMinute)
+            }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Utils/DeviceActivityCenterUtil.swift]: Fixed in 0fc1344 — corrected the fallback: if minute > 0, end = same hour, minute - 1; if minute == 0, end = (hour+23)%24, minute 59. Produces end time exactly 1 minute before start.

File: Foqos/Utils/StrategyManager.swift [hunk 14]
```diff
@@ -1093,6 +1151,78 @@ class StrategyManager: ObservableObject {
...
+  /// Start blocking with a pre-scanned tag (internal helper)
+  private func startWithTag(context: ModelContext, profile: BlockedProfiles, tag: String) {
+    AppBlockerUtil().activateRestrictions(for: BlockedProfiles.getSnapshot(for: profile))
+
+    let session = BlockedProfileSession.createSession(
+      in: context,
+      withTag: tag,
+      withProfile: profile,
+      forceStart: false
+    )
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Utils/StrategyManager.swift]: Fixed in 0fc1344 — `startWithNFCTag` now validates scanned tag against `profile.startNFCTagId` when `specificNFC` is enabled. Returns "This NFC tag doesn't match the one configured for this profile" on mismatch.

File: Foqos/Models/TriggerConfigurationModel.swift [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+      Log.debug(
+        "Trigger validation errors: \(validationErrors.joined(separator: ", ")). "
+          + "Start: manual=\(startTriggers.manual), NFC=\(startTriggers.hasNFC), QR=\(startTriggers.hasQR), schedule=\(startTriggers.schedule), deepLink=\(startTriggers.deepLink). "
+          + "Stop: manual=\(stopConditions.manual), timer=\(stopConditions.timer), NFC=\(stopConditions.anyNFC || stopConditions.specificNFC || stopConditions.sameNFC), "
+          + "QR=\(stopConditions.anyQR || stopConditions.specificQR || stopConditions.sameQR), schedule=\(stopConditions.schedule), deepLink=\(stopConditions.deepLink)",
+        category: .ui
+      )
+    }
+  }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/TriggerConfigurationModel.swift]: Fixed in 0fc1344 — `validate()` now adds errors when specific toggles are on but IDs are nil/empty, and when schedule toggles are on but schedules are nil.

File: Foqos/Utils/StrategyManager.swift [hunk 16]
```diff
@@ -1286,3 +1438,217 @@ extension Notification.Name {
...
+    if conditions.hasNFC {
+      scanOptions.append(.scanNFC)
+    }
+    if conditions.hasQR {
+      scanOptions.append(.scanQR)
+    }
+
+    if scanOptions.isEmpty {
+      return .cannotStop(reason: "This profile can only be stopped when the time runs out")
+    }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Utils/StrategyManager.swift]: Fixed in 0fc1344 — `determineStopAction` now returns contextual messages: timer-only, schedule-only, timer+schedule, deep-link-only, or no manual method configured.

File: Foqos/Models/BlockedProfiles.swift [hunk 17]
```diff
@@ -516,3 +652,123 @@ class BlockedProfiles {
...
+        }
+        if start.manual && stop.anyNFC && stop.timer {
+            return "NFCTimerBlockingStrategy"
+        }
+        if start.manual && stop.anyNFC {
+            return "NFCManualBlockingStrategy"
+        }
+        if start.anyQR && stop.sameQR {
+            return "QRCodeBlockingStrategy"
+        }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/BlockedProfiles.swift]: Fixed in 0fc1344 — now uses `hasNFC`/`hasQR` helpers which include `anyNFC || specificNFC || sameNFC` (and QR equivalents). Also added fallback cases for NFC/QR auto-start profiles.

File: Foqos/Components/BlockedProfileView/StopConditionSelector.swift [hunk 18]
```diff
@@ -0,0 +1,186 @@
...
+  }
+}
+
+private struct StopQRScannerSheet: View {
+  let onScan: (String) -> Void
+
+  var body: some View {
+    Text("QR Scanner for Stop")
+  }
+}
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Components/BlockedProfileView/StopConditionSelector.swift]: REJECT. Stale review — scanners use parent-provided callbacks (`onScanNFCTag`, `onScanQRCode`) which delegate to `BlockedProfileView`'s fully-implemented scanner sheets using `PhysicalReader`.

File: Foqos/Views/HomeView.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Views/HomeView.swift]: REJECT. Stale review — NFC uses `startNFCScan()` → `nfcScanner.scan()` → `startWithNFCTag()`. QR uses `showStartQRScanner` sheet → `LabeledCodeScannerView` → `startWithQRCode()`. Fully implemented, no TODO fallbacks.

File: Foqos/Models/BlockedProfiles.swift [hunk 19]
```diff
@@ -49,6 +49,83 @@ class BlockedProfiles {
...
+    var stopConditions: ProfileStopConditions {
+        get {
+            guard let data = stopConditionsData else { return ProfileStopConditions() }
+            return (try? JSONDecoder().decode(ProfileStopConditions.self, from: data))
+                ?? ProfileStopConditions()
+        }
+        set {
+            stopConditionsData = try? JSONEncoder().encode(newValue)
+        }
+    }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/BlockedProfiles.swift]: REJECT. Already fixed — all JSON encode/decode paths now have `Log.error()` calls with `.sync` category for diagnostics.

File: Foqos/Models/BlockedProfiles.swift [hunk 20]
```diff
@@ -49,6 +49,83 @@ class BlockedProfiles {
...
+    /// Computed property for stop schedule with JSON serialization
+    var stopSchedule: ProfileScheduleTime? {
+        get {
+            guard let data = stopScheduleData else { return nil }
+            return try? JSONDecoder().decode(ProfileScheduleTime.self, from: data)
+        }
+        set {
+            stopScheduleData = newValue.flatMap { try? JSONEncoder().encode($0) }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/BlockedProfiles.swift]: REJECT. Already fixed — same pattern as startTriggers/stopConditions: `Log.error()` on all schedule encode/decode failures.

File: Foqos/Views/HomeView.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Views/HomeView.swift]: REJECT. A `preconditionFailure` in a display-only function would crash the app. Returning a harmless fallback string is the correct defensive approach — `.showPicker` triggers a dialog so this text is never visible anyway.

File: Foqos/Models/BlockedProfiles.swift [hunk 21]
```diff
@@ -516,3 +593,96 @@ class BlockedProfiles {
...
+        if start.manual && stop.anyQR && stop.timer {
+            return "QRTimerBlockingStrategy"
+        }
+        if start.manual && stop.anyQR {
+            return "QRManualBlockingStrategy"
+        }
+
+        // Default to manual
+        return "ManualBlockingStrategy"
+    }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/BlockedProfiles.swift]: ACCEPT. Adding tests for all trigger→strategyId mappings including ShortcutTimerBlockingStrategy edge case.

File: Foqos/Views/HomeView.swift [hunk 22]
```diff
@@ -327,12 +370,61 @@ struct HomeView: View {
...
+    case .startImmediately:
+      strategyManager.toggleBlocking(context: context, activeProfile: profile)
+
+    case .scanNFC:
+      // TODO: Show NFC scanner - for now fall back to toggle
+      strategyManager.toggleBlocking(context: context, activeProfile: profile)
+
+    case .scanQR:
+      // TODO: Show QR scanner - for now fall back to toggle
+      strategyManager.toggleBlocking(context: context, activeProfile: profile)
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Views/HomeView.swift]: REJECT. Duplicate of earlier comment — NFC/QR scanner integration is fully implemented with `startNFCScan()` and QR scanner sheets. No TODO fallbacks remain.

File: Foqos/Models/ProfileStartTriggers.swift (hunk 5 above)
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/ProfileStartTriggers.swift]: REJECT. `deepLink` is intentionally included for forward-compatibility — the field participates in validation, `compatibilityStrategyId` mapping, and `StartAction` determination. Removing it would require a schema change to re-add later.

File: Foqos/Models/TriggerConfigurationModel.swift [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+    }
+    if stopConditions.specificQR && (stopQRCodeId == nil || stopQRCodeId?.isEmpty == true) {
+      errors.append("Scan a QR code to use as the stop condition")
+    }
+    if startTriggers.schedule && startSchedule == nil {
+      errors.append("Configure a start schedule")
+    }
+    if stopConditions.schedule && stopSchedule == nil {
+      errors.append("Configure a stop schedule")
+    }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/TriggerConfigurationModel.swift]: ACCEPT. Confirmed: validation checks `nil` but not `isActive`. A schedule with empty days passes. Adding `isActive` check.

File: Foqos/Views/BlockedProfileView.swift [hunk 24]
```diff
@@ -337,6 +365,30 @@ struct BlockedProfileView: View {
...
+                    disabled: isBlocking || (isManagedProfile && !isUnlockedForEditing),
+                    onConditionChange: {
+                        triggerConfig.stopConditionsDidChange()
+                    },
+                    onScanNFCTag: {
+                        nfcScanner.onTagScanned = { tag in
+                            triggerConfig.stopNFCTagId = tag.id
+                        }
+                        nfcScanner.scan(profileName: "stop trigger")
+                    },
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Views/BlockedProfileView.swift]: ACCEPT. Confirmed: stop-trigger NFC scan sets `onTagScanned` but not `onError`. HomeView does set both. Adding `onError` handler.

File: Foqos/Views/BlockedProfileView.swift [hunk 25]
```diff
@@ -310,20 +329,29 @@ struct BlockedProfileView: View {
...
+                    disabled: isBlocking || (isManagedProfile && !isUnlockedForEditing),
+                    onTriggerChange: {
+                        triggerConfig.startTriggersDidChange()
+                    },
+                    onScanNFCTag: {
+                        nfcScanner.onTagScanned = { tag in
+                            triggerConfig.startNFCTagId = tag.id
+                        }
+                        nfcScanner.scan(profileName: "start trigger")
+                    },
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Views/BlockedProfileView.swift]: ACCEPT. Same gap as stop-trigger: start-trigger NFC scan also missing `onError`. Adding handler.

File: Foqos/Models/TriggerPickerOptions.swift [hunk 26]
```diff
@@ -0,0 +1,140 @@
...
+    case .same: return "Same tag"
+    case .specific: return "Specific tag"
+    }
+  }
+
+  static func from(_ conditions: ProfileStopConditions) -> NFCStopOption {
+    if conditions.anyNFC { return .any }
+    if conditions.sameNFC { return .same }
+    if conditions.specificNFC { return .specific }
+    return .none
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/TriggerPickerOptions.swift]: Accepted. Reordered NFCStopOption.from() to specific > same > any, matching canStop enforcement precedence.
